with-crm-components/src/app/extensions/StageTrackerExtension.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect } from 'react';
+import {
+  hubspot,
+  Button,
+  Form,
+  Select,
+  Accordion,
+  Flex,
+  Heading,
+  Alert,
+} from '@hubspot/ui-extensions';
+import {
+  CrmStageTracker,
+  CrmAssociationPivot,
+} from '@hubspot/ui-extensions/crm';
+
+// Define the extension to be run within the Hubspot CRM
+hubspot.extend(({ context, actions, runServerlessFunction }) => (
+  <Extension
+    context={context}
+    runServerless={runServerlessFunction}
+    fetchCrmObjectProperties={actions.fetchCrmObjectProperties}
+    addAlert={actions.addAlert}
+  />
+));
+
+const Extension = ({
+  context,
+  runServerless,
+  fetchCrmObjectProperties,
+  addAlert,
+}) => {
+  const [stage, setStage] = useState(null);
+  const [showProperties, setShowProperties] = useState(true);
+  const [dealId, setDealId] = useState(null);
+  const [dealname, setDealname] = useState(null);
+  const [error, setError] = useState('');
+
+  const stageToPropertiesMap = {
+    appointmentscheduled: [
+      'dealname',
+      'engagements_last_meeting_booked',
+      'dealtype',
+    ],
+    qualifiedtobuy: ['hubspot_owner_id', 'amount', 'dealtype', 'hs_priority'],
+    presentationscheduled: [
+      'hs_priority',
+      'hs_deal_stage_probability',
+      'hs_forecast_amount',
+    ],
+    decisionmakerboughtin: [
+      'hs_deal_stage_probability',
+      'hs_tcv',
+      'amount',
+      'notes_last_contacted',
+    ],
+    contractsent: ['createdate', 'hs_acv', 'hs_deal_stage_probability'],
+    closedwon: ['closed_won_reason', 'closedate', 'amount'],
+    closedlost: ['closedate', 'closed_lost_reason', 'amount'],
+  };
+  const options = [
+    { value: 'appointmentscheduled', label: 'Appointment Scheduled' },
+    { value: 'qualifiedtobuy', label: 'Qualified to Buy' },
+    { value: 'presentationscheduled', label: 'Presentation Scheduled' },
+    { value: 'decisionmakerboughtin', label: 'Decision Maker Bought In' },
+    { value: 'contractsent', label: 'Contract Sent' },
+    { value: 'closedwon', label: 'Closed Won' },
+    { value: 'closedlost', label: 'Closed Lost' },
+  ];
+
+  useEffect(() => {
+    const fetchProperties = () => {
+      fetchCrmObjectProperties(['dealname', 'dealstage', 'hs_object_id']).then(
+        (properties) => {
+          setStage(properties.dealstage);
+          setDealId(properties.hs_object_id);
+          setDealname(properties.dealname);
+        }
+      );
+    };
+    fetchProperties();
+  }, [stage]);
+
+  const handleStageChange = (newStage) => {
+    runServerless({
+      name: 'update',
```

Comment:
Nit: can we make this name a little more specific? Something like "updateDeal" to make it a little easier to follow through the code?